calicoctl/calicoctl/commands/convert.go

```diff
@@ -239,3 +258,56 @@ func getTypeConverter(resKind string) (converters.Converter, error) {
 		return nil, fmt.Errorf("conversion for the resource type '%s' is not supported", resKind)
 	}
 }
+
+func unpackResourceLists(convRes []unversioned.Resource) ([]unversioned.Resource, error) {
+	var unpackedConvRes []unversioned.Resource
+	for i, convResource := range convRes {
+		if strings.EqualFold(convResource.GetTypeMetadata().Kind, resourceloader.KindK8sListV1) && strings.EqualFold(convResource.GetTypeMetadata().APIVersion, resourceloader.VersionK8sListV1) {
+			// Remove list from convRes
+			convRes[i] = convRes[len(convRes)-1]
+			convRes = convRes[:len(convRes)-1]
+
+			k8sNPList, ok := convResource.(*resourceloader.K8sNetworkPolicyList)
+			if !ok {
+				return nil, fmt.Errorf("failed to convert resource to K8sNetworkPolicyList")
+			}
+
+			var listResources []unversioned.Resource
+			for _, item := range k8sNPList.Items {
```

Comment:
you should be able to skip this intermediate list right?

e.g.,

```go
for _, item := range k8sNPList.Items {
  unpakcedConvRes = append(unpackedConvRes, item)
```

Or maybe you need an interface assertion?

```go
for _, item := range k8sNPList.Items {
  unpakcedConvRes = append(unpackedConvRes, unversioned.Resource(item))
```

Reply:
That's much better, thanks!